Fix/549 release table updates

hoshmn opened the pull request:
> Remove hover/highlight functionality from other tables on the site which are not clickable

@Lane / @ekmason the one table I can find that this seems to apply to is the facility table, where when a row is hovered it gets a highlighted background and its state name gets underlined to indicate the user can click the state name to view that state page. Should I remove the row highlighting there, but leave the state underlining?

I made the table very slightly wider (1rem) and reallocated more of the width to the release details column. With this done, it seems like we could just show the whole release details value rather than trying to cut it off with an ellipsis after 2 lines (which can't be done simply with CSS). Because the cell content text can never be more than a known length (it is generated by aggregating checked options in their source spreadsheet) we know the rows will never grow much -- see the following screenshot, where the Franklin row has the longest possible content (spanning 4 rows):

<img width="940" alt="Screen Shot 2021-09-27 at 6 54 17 PM" src="https://user-images.githubusercontent.com/18331999/135010597-259b1cca-2316-4f0d-9b1f-75e5fdd80ce6.png">

If we really don't want the rows to shift slightly when a user clicks through pages of the table and there happen to be release details cells that span more than 2 rows, one option would be to set the height of all rows to a height that allows the 4 rows to show. The following shows the rows set at about 18px taller than they currently are (which cuts slightly into the padding of rows that have the very longest release details, but we could make them slightly taller if we don't want to cut into that padding):

<img width="1189" alt="Screen Shot 2021-09-27 at 6 14 22 PM" src="https://user-images.githubusercontent.com/18331999/135010595-f931689c-1add-4398-85fb-5151c2a3c9f6.png">

Here's what the tooltips look like:

<img width="828" alt="Screen Shot 2021-09-27 at 7 09 04 PM" src="https://user-images.githubusercontent.com/18331999/135011518-7f205447-b172-4d5d-b2ad-36e1e9bfcd0a.png">

Lastly, certain rows don't have any facility name mentioned. I figured I'd put in a default to show in those cases, maybe "Multiple facilities" (not sure if that's what is implied by the facility missing?).

Thoughts?

fixes #549

hoshmn commented on src/components/states/sections/ReleasesTable.js:
```diff
@@ -46,10 +52,26 @@ const ReleasesTable = ({ classes, data, lang, ...props }) => {
+          return (
+            <>
+              <Typography className={classes.facilityName} variant="body1">
+                {facility || "Multiple facilities"}
```
todo: add to CMS if we do go with a default here

ekmason commented:
> Should I remove the row highlighting there, but leave the state underlining?

Since this functionality was pretty recently implemented (this was part of Phase 2, right?) and Liz approved, I'd lean toward keeping it - I'll defer to @Lane on this of course! 

> Because the cell content text can never be more than a known length (it is generated by aggregating checked options in their source spreadsheet) we know the rows will never grow much

Is this because the field is comprised of members of a closed option set? I'll check with Liz on this, but couldn't that option set grow if UCLA thought it necessary? 

>Lastly, certain rows don't have any facility name mentioned. I figured I'd put in a default to show in those cases, maybe "Multiple facilities" (not sure if that's what is implied by the facility missing?).

I'll raise this to Liz today to get a sense of what behavior she'd like to see here. 

--

I'm noticing that as the table gets longer, the legend shifts down with it - thus distancing it from the map. Would it be possible to keep the legend just beneath the map despite the length of the table?

Lane commented:
thanks @hoshmn and @ekmason!  a few additional notes.

>  Should I remove the row highlighting there, but leave the state underlining?

yes, the row isn't clickable so it shouldn't have a hover state.  the state link is the clickable item, so it should have a hover state when hovered, but it should not have a hover state when the row is hovered.

> show full release details in table

i'm not a fan of the variable height rows because they cause jumping between pages, but given it's not a huge jump it's fine with me.

> facility names

can we adjust styling of these to match other table rows where we have a primary + secondary value:

![image](https://user-images.githubusercontent.com/21034/135323854-7ea60e34-42a1-49ca-a73c-b09a8e1dd46a.png)

hoshmn commented:
thanks for the feedback both @Lane & @ekmason ! i made those adjustments. Eliana good catch on the legend getting further than ideal from the state -- unfortunately i don't this is a product of the table getting taller but rather happens when a state has a higher width:height ratio and so it doesn't extend to the bottom of its allotted svg container, leaving a gap between it and the legend. i don't think there's any easy fix for that :\

Lane commented:
@hoshmn can you set `position: sticky` on the map wrapper to keep the map in view within the section?  for some states, when you get to the table part of the state can be scrolled out of view:

current behaviour:

![image](https://user-images.githubusercontent.com/21034/135685798-ee1e7ff1-5191-42e8-a06f-5f70ab01b35d.png)


desired behaviour:

![2021-10-01 14 03 45](https://user-images.githubusercontent.com/21034/135685958-b768f685-603f-4ae8-ae57-c3f8bafa74fb.gif)

Lane commented:
@hoshmn thanks, sticky looks good!

noticing the tooltip is coming up above the hovered row instead of to the right on the preview:

![image](https://user-images.githubusercontent.com/21034/135935846-cc6c641a-29ab-4f37-849c-cd022268cf4c.png)

also noticing that there is no hovered state on the table rows.  we want to keep the hover state on rows that are clickable (have actions tied to them) but remove from tables where clicking on the row doesn't do anything.

hoshmn commented:
@Lane 
> noticing the tooltip is coming up above the hovered row instead of to the right on the preview:

moved it back - moved it to "top" because that's where they go for hovered facility names for the HomeTable
<img width="1268" alt="Screen Shot 2021-10-04 at 4 52 15 PM" src="https://user-images.githubusercontent.com/18331999/135939484-b14e2597-7d0c-47a9-92eb-c89fb10e4b43.png">
and, if set to "right" and there isn't room to the right, they obscure the beginning of the row
<img width="733" alt="Screen Shot 2021-10-04 at 4 53 50 PM" src="https://user-images.githubusercontent.com/18331999/135939599-d41d3e10-5eee-4806-bfd9-b895061c0f9a.png">
but I'm fine with whatever you think is best


> also noticing that there is no hovered state on the table rows. we want to keep the hover state on rows that are clickable (have actions tied to them) but remove from tables where clicking on the row doesn't do anything.

whoops! added the hover state back for tables passed an `onRowClick` handler

Lane commented:
gotcha, we want to keep these "click row" tooltips out of the data area, so right works best i think.

not too concerned about the placement on the smaller resolutions because those will mainly be touch devices without tooltips. 

thanks for those changes, merging in!
